### ISSUE TITLE
Many redraws result in a "texture unit out of range" error

### ISSUE THREAD

**Issue by plankter** [flekschas/regl-scatterplot#22]
There is a very strange issue, which was quite hard to track. In my app after exactly 32 scatterplot updates involving setting `backgroundImage` property, there is always a crash with the exception: `WebGL: INVALID_ENUM: activeTexture: texture unit out of range`.

I created a demo using existing `texture-background.html` example to show the problem. Here it is also crashing exactly after 32 ticks: https://github.com/plankter/regl-scatterplot-texture-background-issue (please check the console).

Also, i don't know if it is by design, but setting `backgroundImage` property doesn't update a plot if there is no `scatterplot.draw()` call.

**Comment by flekschas** [comment]
The issue does not seem to be related to regl-scatterplot. When you properly destroy previous texture objects, I don't see any issues. I.e., the following code works just fine:

```javascript
let imgId = 0;
function setBackgroundImage() {
  const url = `https://i.picsum.photos/id/${imgId}/64/64.jpg`;
  scatterplot.createTextureFromUrl(url).then((texture) => {
    const prevBackgroundImage = scatterplot.get('backgroundImage');

    scatterplot.set({ backgroundImage: texture });

    if (prevBackgroundImage) {
      prevBackgroundImage.destroy();
    }

    console.log('tick', imgId);

    imgId++;

    setTimeout(setBackgroundImage, 500);
  });
}

setBackgroundImage();
```

Note that there were a couple of issues with the previous implementation:
1. `setTimeout` is brittle as network connections can break or slow down. It's better to wait for the image to load and then call it a _tick_.
2. There's no need anymore to manually create an image. You can just use `scatterplot.createTextureFromUrl(url)` which returns a promise once the image is loaded and turned into a texture object.

In case you wonder why regl-scatterplot doesn't just destroy the texture itself: I don't know what else you do with the texture object. It could be used elsewhere. What I can try to add in a future version is to track if the user passed in a string or texture object. In case the user passed in a string, it's likely that the texture object isn't used elsewhere so I could opt for destroying it. I'll think about that.

I'll close the issue for now as I think you just exceeded the [maximum number of texture units](http://learnwebgl.brown37.net/10_surface_properties/texture_mapping_images.html#hardware-overview). If I am mistaken please reopen the issue.

**Comment by plankter** [comment]
Thank you for the explanation. Unfortunately, i have to reopen the issue.

First, i found out it has nothing to do with `backgroundImage`. I replaced the timer call method in the demo with the following:

```js
setInterval(() => {
  scatterplot.draw([]);
  console.log('tick');
}, 1000);
```

and it still crashes with the same error after exactly 32 ticks. Very awkward.

I cannot use `createTextureFromUrl` method because i got images using POST requests (length of GET requests sometimes is not enough as i send a lot of parameters) and also there is some JWT auth involved.

BTW, disposing images via `destroy()` didn't help either.

**Comment by flekschas** [comment]
Thanks for your further investigation. I realized I did the same mistake: a texture object wasn't properly destroyed before it was recreated.

fae273436ae6bbf9848d0018bc34ce56265e1958 should fix the issue.

Can you test the example with the latest version of the master? If it's fixed I'll release a new version soon.

**Comment by plankter** [comment]
Thank you! Seems like issue is fixed.

**Comment by flekschas** [comment]
Fixed in fae2734
